packages/react-dom/src/__tests__/ReactUpdates-test.js

```diff
@@ -1362,32 +1362,27 @@ describe('ReactUpdates', () => {
   });
 
   it('does not fall into an infinite error loop', () => {
-    function BadRender() {
-      throw new Error('error');
-    }
-
-    class ErrorBoundary extends React.Component {
-      componentDidCatch() {
-        this.props.parent.remount();
+    class BadMount extends React.Component {
+      componentDidMount() {
+        throw new Error('error');
       }
       render() {
-        return <BadRender />;
+        return null;
       }
     }
 
-    class NonTerminating extends React.Component {
-      state = {step: 0};
-      remount() {
-        this.setState(state => ({step: state.step + 1}));
+    class ErrorBoundary extends React.Component {
+      componentDidCatch() {
+        // Noop
       }
       render() {
-        return <ErrorBoundary key={this.state.step} parent={this} />;
+        return <BadMount />;
       }
     }
 
     const container = document.createElement('div');
     expect(() => {
-      ReactDOM.render(<NonTerminating />, container);
+      ReactDOM.render(<ErrorBoundary />, container);
     }).toThrow('Maximum');
```

Comment:
Can you explain these changes? I’m not quite following.

Reply:
Previous test case works fine. I updated it so the error is thrown in the render phase instead of the commit phase, since we already have commit phase coverage. But I suppose both test cases are useful, so I'll add the old one back too.